source/_integrations/onvif.markdown

```diff
@@ -88,4 +90,54 @@ If your ONVIF camera supports PTZ, you will be able to pan, tilt or zoom your ca
 | `move_mode` | PTZ moving mode. Allowed values: `ContinuousMove`, `RelativeMove`, `AbsoluteMove`, `GotoPreset`, `Stop`. Default :`RelativeMove` |
 | `continuous_duration` | Set ContinuousMove delay in seconds before stopping the move. Allowed values: floating point numbers or integer. Default : 0.5 |
 
-If you are running into trouble with this sensor, please refer to the [Troubleshooting section](/integrations/ffmpeg/#troubleshooting).
+### Service `onvif.aux`
+
+If your ONVIF camera supports PTZ auxiliary commands (e.g. for running the wiper or turning on the heater), you will be able to run them with this service.
+
+| Service data attribute | Description |
+| -----------------------| ----------- |
+| `entity_id` | String or list of strings that point at `entity_id`s of cameras. Use `entity_id: all` to target all. |
+| `cmd` | Command to send (e.g. `tt:Wiper|On`). Options depend on camera capability. |
+
+### Service `onvif.set_imaging_settings`
+
+Allos you to set various Imaging Settings as defined in [The ONVIF Imaging
+Service Spec](https://www.onvif.org/specs/srv/img/ONVIF-Imaging-Service-Spec-v210.pdf). 
+
+| Service data attribute | Description |
+| -----------------------| ----------- |
+| `entity_id` | String or list of strings that point at `entity_id`s of cameras. Use `entity_id: all` to target all. |
+| `settings` | Command to send (e.g. `tt:Wiper|On`). Options depend on camera capability. |
+
+Valid setting keys depend on your camera capabilities, but may include:
+
+* `BacklightCompensation`
+* `Brightness`
+* `ColorSaturation`
+* `Contrast`
+* `Exposure`
+* `Focus`
+* `IrCutFilter`
+* `Sharpness`
+* `WideDynamicRange`
+* `WhiteBalance`
```

Comment:
Why aren't these just entities/controls?

Reply:
The initial reasons this is a service are:

* the related Point-tilt-zoom settings that the existing integration has are also a service, so I was attempting consistency with that 
* some of the values have slightly complicated data structs and take a number of options that this PR is leaving to the user (who then is therefore expected to be somewhat of a ONVIF power-user I guess).

Making them entities/controls would indeed make things more powerful and easy, so that's a fair point. 

Looking at other related examples, I see controls like this often included as data attrs on services, such as `light.turn_on`. I can try to make them controls like that in this PR if that would be preferred. 

Just to clarify, I still need service(s) though, right? In order to publish the control values to the camera, I need a service that can accept the various parameters? Maybe a number of services, such as `turn_wiper_on`, `turn_wiper_off`, `turn_ir_lamp_on`, `turn_ir_lamp_off`, `update_exposure_settings` etc?

Reply:
> some of the values have slightly complicated data structs and take a number of options that this PR is leaving to the user (who then is therefore expected to be somewhat of a ONVIF power-user I guess).

Our goal should be to become more user-friendly. This is completely defeating those goals.

> Looking at other related examples, I see controls like this often included as data attrs on services, such as light.turn_on. I can try to make them controls like that in this PR if that would be preferred.

That is not related, as those are feature of the light entity component. In this PR nothing is related to entity component features.

> Just to clarify, I still need service(s) though, right?

No? Just entities.

Reply:
I updated them to be entities in the base PR. Hoping to get some feedback on whether or not that is the expected route, and if so I'll continue making more entities for more of the controls.

https://github.com/home-assistant/core/pull/84317